options/options.go

```diff
@@ -202,6 +199,51 @@ func interpretOptions(lenient bool, file file, handler *reporter.Handler) (Index
 	return interp.index, nil
 }
 
+func resolveDescriptor[T protoreflect.Descriptor](res linker.Resolver, name string) T {
+	var zero T
+	if res == nil {
+		return zero
+	}
+	if len(name) > 0 && name[0] == '.' {
+		name = name[1:]
+	}
+	desc, _ := res.FindDescriptorByName(protoreflect.FullName(name))
+	typedDesc, ok := desc.(T)
+	if ok {
+		return typedDesc
+	}
+	return zero
+}
+
+func (interp *interpreter) resolveExtensionType(name string) protoreflect.ExtensionTypeDescriptor {
+	if interp.resolver == nil {
+		return nil
+	}
+	if len(name) > 0 && name[0] == '.' {
+		name = name[1:]
+	}
+	ext, _ := interp.resolver.FindExtensionByName(protoreflect.FullName(name))
```

Comment:
Do we need to handle the error here?

Reply:
We _at the least_ need to handle if `ext` is nil, to avoid a potential panic on the next line. I actually caught that in [another branch](https://github.com/bufbuild/protocompile/pull/110/files#diff-76f121a3a81c8945dc0e0c81a4b95e941d252522eba8704a1aa6f6a1722a8b68R227-R229) but forgot to apply the change here. Let me update this...